QUICKSTART.md

```diff
@@ -0,0 +1,180 @@
+Dry run, nothing changed.
+Push done.
+```
+### kbc push
+Finally, to sync local changes into the remote project call `kbc push`.
+```
+➜  my-kbc-project kbc push
+Plan for "push" operation:
+	CH C main/extractor/ex-generic-v2/729819493-some-api | changed: configuration
+Push done.
+```
+
+## Create new configurations
+Sometimes you need to create a new configuration from the local directory a sync it to remote project. In such you case you need to create files following the [outlined directory structure](README.md#directory-structure). After that use `kbc persist` command to prepare configuration(to automatically fill-up ids and additional info) and then call `kbc push` command. 
+### Prepare configuration files
+Let's say we want to create new `ex-generic-v2` confiuration named `wiki`. Under `main/extractor/ex-generic-v2/` we create a directory `wiki` where we create following files:
+- `config.json` - with the following json content:
+```
+{
+    "api": {
+        "baseUrl": "https:wiki.com"        
+    }
+}
+```
+ - `meta.json` - with the following json content:
+```
+{
+  "name": "extract wiki.com",
+  "description": ""
+}
+```
+ 
+The final file structure would look as follows:
+```
+ ➜  my-kbc-project tree
+.
+└── main
+    ├── extractor
+    │   ├── ex-generic-v2
+    │   │   ├── 729814342-my-generic-api
+    │   │   │   ├── config.json
+    │   │   │   └── meta.json
+    │   │   ├── 729819493-some-api
+    │   │   │   ├── config.json
+    │   │   │   └── meta.json
+    │   │   └── wiki
+    │   │       ├── config.json
+    │   │       └── meta.json
+```
+### kbc persist
+Call `kbc persist` command to finalize the wiki configuration files.
+```
+ ➜  my-kbc-project kbc persist
+New objects:
+	+ C 731114520 main/extractor/ex-generic-v2/wiki
+Renamed objects:
+	- main/extractor/ex-generic-v2/{wiki -> 731114520-extract-wiki-com}
+Persist done.
+```
+Note that the `wiki` directory has been renamed `731114520-extract-wiki-com` to meet the naming conventions.
+ 
+### kbc push --dry-run
+Call `kbc push --dry-run` to preview changes:
+```
+➜  my-kbc-project kbc push --dry-run
+Plan for "push" operation:
+	+  C main/extractor/ex-generic-v2/731114520-extract-wiki-com
+Dry run, nothing changed.
+```
+### kbc push
+Finally, to sync local changes into the remote project call `kbc push`.
+```
+➜  my-kbc-project kbc push
+Plan for "push" operation:
+	+  C main/extractor/ex-generic-v2/731114520-extract-wiki-com
+Push done.
+``` 
+
+## Sync remote changes to the local directory
+Sometimes you create a new or change an existing configuration in the remote project.
+### kbc diff --details
+Preview changes between remote project and local directory call `kbc diff --details`.
+```
+➜  my-kbc-project kbc diff --details
+CH changed
+-  remote state
++  local state
+
+Diff:
+CH C main/extractor/ex-generic-v2/729819493-some-api
+	"configuration":
+		"newRemoteProperty":
+		- newRemoteValue
```

Comment:
tento diff mi pride podozrivy - neni tu bug?
som vlastne len na UI pridal property vid diff:
![image](https://user-images.githubusercontent.com/1412120/126185642-36dd07d2-dc4d-4b0d-a258-763b9dcfdd07.png)

Reply:
@kacurez ja tam nevidim chybu, .. `-` znamena, ze `to chyba v local, je iba v remote` ... ked das `push`, tak sa to z `remote` stavu zmaze.

Mozeme tam dat nejaku inu znacku ako `-`, ale velmi neviem aku, ... mozno sipku hore/dole, ale to neviem ci kazdy terminal vykresli.

Reply:
ja som cakal ze ten diff bude vyzerat podobne ako na UI(vid obrazok vyssie), ale vlastne tu sa riesi zmena jak zo strany local directory tak aj remote project. Tak ako to popisujes mi to dava zmysel.

Reply:
jj, nie je to diff medzi dvoma verziami ... jedine fakt asi pouzit ine znacky, ... v jednom diffe mozes mat nove hodnoty v `remote` aj `local` state, a musia byt nejak odlisene, ... nestaci teda iba `+` pre nove hodnoty.

Reply:
mozno by bolo lepsie `R` a `L`, ak ti to dava vacsi zmysel, tak to mozem upravit.

Reply:
by som to teraz nechal tak ako je a riesil to az ked nato pride vacsi feedback